class1/README.md

```diff
@@ -25,12 +25,100 @@ Helping us this semester with Ninja duties are:
+
+Together TCP/IP is known as the Internet Protocol Suite. There are many other protocols that you can use to communicate over the internet, but this suite is the backbone of most connections, and as we'll see, HTTP. That's about all you'll need to know about TCP/IP for this class, so let's pay close attention to this next section which will be much more relevant throughout the semester.
+
+###HTTP
+The **HyperText Transfer Protocol** (**HTTP**) is the popular protocol used for exchanging resources - like web pages, images, and even raw data - over the web. Yes, it's yet another protocol and it lays out the format of data to be sent out over TCP. HTTP is designed for a **client-server** architecture, in which a client makes a **request** to a server, to which the server sends a **response** (yes, these are technical terms, that's why they're **bolded**). Requests and responses have a simple plaintext format that is divided into a **header** section, and a **body** section.
+
+#####HTTP Requests
+To understand an HTTP request, let's see what one looks like!
+
+```
+    GET /wiki/List_of_HTTP_header_fields HTTP/1.1
+    Host: en.wikipedia.org:80
+    Connection: keep-alive
+    Cache-Control: max-age=0
+    Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
+    User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36
+    Referer: https://www.google.com/
+    Accept-Encoding: gzip, deflate, sdch
+    Accept-Language: en-US,en;q=0.8
+```
+
+Now at first it might look a bit intimidating, but if we read through it carefully, we'll see there's nothing to be afraid of. Let's start at the top.
+
+```
+GET /wiki/List_of_HTTP_header_fields HTTP/1.1
+```
+
+This line actually tells us 3 things. 1) We want to `GET` the resource located at the URL 2) `/wiki/List_of_HTTP_header_fields`, and we are expecting this transaction to use 3) version 1.1 of the HTTP protocol. The next line is the start of the **field-value** part of the header. As an example, the first field-value pair,
+
+```
+Host: en.wikipedia.org:80
+```
+
+is required under version 1.1 and tells the server the domain name and port number it is expecting to connect to. This is important for scenarios where a single server can be running multiple VMs, each supporting a different domain, and the hardware needs to know which VM to send the request to. The rest of the field-values are there to give special instructions to the server in the event that it needs that information. You won't really care about that stuff for a while, and when you need it 
```

Comment:
This section appears to be missing an end. Also, maybe user-agent is worth a brief explanation since people have likely heard of it?